cmd/buildah/bud.go

```diff
@@ -107,8 +107,10 @@ func budCmd(c *cobra.Command, inputArgs []string, iopts budOptions) error {
 			tags = tags[1:]
 		}
 	}
-	if err := auth.CheckAuthFile(iopts.BudResults.Authfile); err != nil {
-		return err
+	if c.Flag("authfile").Changed {
```

Comment:
could we add:
` || os.Getenv("REGISTRY_AUTH_FILE"); authfile == "" `

to these, and I think this addresses my concerns and @vrothberg's